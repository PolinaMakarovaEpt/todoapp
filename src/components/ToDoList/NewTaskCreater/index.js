import React, { Component } from 'react';
import './index.css';

class AddTask extends Component {

  constructor(props) {
    super(props);

    this.state = {
      input: ''
    };
  }

  onCreate(e) {

    if (e.key === 'Enter') {
      if (this.state.input !== '') {

        const newTask = {
          title: this.state.input,
          checked: false,
          id: Date.now()
        };

        this.props.onCreate(newTask);
        this.setState({
          input: ''
        });
      }
    }
  }

  handleChange(e) {

    this.setState({
      input: e.target.value
    });
  }

  handleClick(e) {

    e.preventDefault();

    if (this.state.input !== '') {

      const newTask = {
        title: this.state.input,
        checked: false,
        id: Date.now()
      }

      this.props.onClick(newTask);
      this.setState({
        input: ''
      });
    }
  }

  render() {

    return (
      <div className="add-task">

        <input
          value={this.state.input}
          autoFocus placeholder="Введите дело"
          onKeyPress={this.onCreate.bind(this)}
          onChange={this.handleChange.bind(this)} />

        <button id="add" onClick={this.handleClick.bind(this)}>Добавить</button>

      </div>
    );
  }
}

export default AddTask;
