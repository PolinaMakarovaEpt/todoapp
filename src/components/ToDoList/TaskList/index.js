import React from 'react';
import Task from './Task/index';

function TasksList(props) {

  return (
    <div className="tasks">
      {props.tasks.map((task) => {
        return <Task task={task}
          onDelete={props.onDelete}
          onCheck={props.onCheck}
          key={task.id} />
      })
      }
    </div>
  );
}

export default TasksList;
