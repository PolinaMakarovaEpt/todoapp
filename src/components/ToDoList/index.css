html {
    font-size: 20px;
}

hr {
    padding: 0%;
    margin: 0%;
}

h3 {
    margin: 1rem 0 0rem 0.5rem;
}

.delete-completed, .delete, .task, .checked, button {
    cursor: pointer;
    word-break: break-all;
}

.container {
    display: block;
    width: 50%;
    padding: 0;
    margin: 0 auto;
    margin-top: 2rem;
    border: 2px solid #4c2761;
    border-radius: 10px;
}

.date {
    position: relative;
    width: 100%;
    padding: 0.4rem 0 0.5rem 0 ;
    background-color: #4c2761;
    border-radius: 7px 7px 0 0;
    font-size: 25px;
    text-align: center;
    color: white;
    border-bottom: 1px solid white;
}

/* 1 запрос */
@media only screen and  (max-width: 1168px) and (min-width: 800px) {
    .container {
        width: 90%;
    }
}

/* 2 запрос */
@media only screen and  (max-width: 800px) and (min-width: 510px) {
    .container{
        width: 98%;
    }
}

/* 3 запрос */ 
@media only screen and  (max-width: 510px) {
    .container{
        width: 100%;
        border: none;
    }
    input {
        width: 50%;
        font-size: 18px;
    }
    #add {
        font-size: 18px;
    }
    .footer {
        position: relative;
        font-size: 15px;
        top: 0.5rem;
    }
    .footer-buttons {
        width: 30%;
    }
    .footer button {
        position: relative;
        font-size: 13px;
        bottom: 0.5rem;
    }
    .delete-completed {
        position: relative;
        bottom: 0.5rem;
        right: 1rem;
        width: 20%;
        word-break: keep-all;
        margin: 0 0.2rem;
    }
    .active-items {
        position: relative;
        bottom: 0.5rem;
        margin-right: 0.2rem;
    }
}
  