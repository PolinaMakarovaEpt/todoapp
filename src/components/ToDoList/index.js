import React, { Component } from 'react';
import './index.css';
import AddTask from './NewTaskCreater/index';
import TasksList from './TaskList/index';
import Footer from './Footer/index';

class ToDoList extends Component {

  constructor() {
    super();

    this.state = {
      tasks: [],
      filter: "all"
    }
  }

  createNewTask(task) {

    this.setState({
      tasks: [...this.state.tasks, task]
    });
  }

  handleClick(task) {
    this.setState({
      tasks: [...this.state.tasks, task]
    });
  }

  deleteTask(taskId) {
    const ifDelete = window.confirm('Удалить задачу?')

    if (ifDelete) {
      const newTasksList = this.state.tasks.filter((t) => {
        return t.id !== taskId;
      });

      this.setState({
        tasks: newTasksList
      });
    }
  }

  checkTask(task) {
    const newTasksList = [...this.state.tasks]

    newTasksList.forEach((t) => {
      if (t.id === task.id) {
        t.checked = task.checked;
        return;
      }
    });

    this.setState({
      tasks: newTasksList
    });
  }

  changeFilter(filterValue) {
    this.setState({
      filter: filterValue
    });
  }

  clearCompleted() {
    const ifClear = window.confirm('Удалить все завершенные дела?')

    if (ifClear) {
      this.setState({
        tasks: this.state.tasks.filter((t) => !t.checked)
      });
    }
  }

  saveTodos = () => {
    const tasks = this.state.tasks;
    localStorage.setItem("storage", JSON.stringify(tasks));
  }

  componentDidUpdate() {
    this.saveTodos();
  }

  componentDidMount() {
    const data = JSON.parse(localStorage.getItem("storage"));
    const tasks = (data && [...data]) || [];
    this.setState({ tasks });
  }

  render() {

    let filteredTasks = [];

    if (this.state.filter === 'all') filteredTasks = this.state.tasks;
    if (this.state.filter === 'active') filteredTasks = this.state.tasks.filter((t) => !t.checked);
    if (this.state.filter === 'completed') filteredTasks = this.state.tasks.filter((t) => t.checked);

    return (
      <div className="todolist">
        <h3>{this.state.tasks.filter((t) => !t.checked).length === 0 ? "Все дела завершены! Добавьте новое" : "Добавить новое дело"}</h3>
        <AddTask
          onCreate={this.createNewTask.bind(this)}
          onClick={this.handleClick.bind(this)} />

        <TasksList
          tasks={filteredTasks}
          onDelete={this.deleteTask.bind(this)}
          onCheck={this.checkTask.bind(this)} />

        <Footer
          tasks={this.state.tasks}
          filter={this.state.filter}
          onFilterChanged={this.changeFilter.bind(this)}
          onClearCompleted={this.clearCompleted.bind(this)}
        />
      </div>
    );
  }
}

export default ToDoList;
