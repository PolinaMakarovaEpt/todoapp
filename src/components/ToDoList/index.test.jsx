import React from "react";
import ToDoList from "./index";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

it("matches the snapshot", () => {
    const tree = shallow(<ToDoList />);
    expect(toJson(tree)).toMatchSnapshot();
  });
