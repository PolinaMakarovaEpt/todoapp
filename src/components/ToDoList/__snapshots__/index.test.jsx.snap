// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="todolist"
>
  <h3>
    Все дела завершены! Добавьте новое
  </h3>
  <AddTask
    onClick={[Function]}
    onCreate={[Function]}
  />
  <TasksList
    onCheck={[Function]}
    onDelete={[Function]}
    tasks={Array []}
  />
  <Footer
    filter="all"
    onClearCompleted={[Function]}
    onFilterChanged={[Function]}
    tasks={Array []}
  />
</div>
`;
